### PR TITLE
Update the contracts to solidity 0.5.0

### PR DESCRIPTION
**Pull request by elopio**
Requires #1492

### CODE REVIEW

File: contracts/utils/ReentrancyGuard.sol [hunk 1]
```diff
@@ -1,4 +1,4 @@
-pragma solidity ^0.4.24;
+pragma solidity >0.4.24;
```
Comment (axic) [inline contracts/utils/ReentrancyGuard.sol]: I suggest to use `>0.4.24 <0.6.0`

**Comment by elopio** [comment]
@jtakalai this needs a little more work, because there are still tests failing.

**Comment by axic** [comment]
Any chance to merge this soon?

**Comment by nventuro** [comment]
Closing in favor of https://github.com/OpenZeppelin/openzeppelin-solidity/pull/1510 - I did take some changes from here, but I believe the merge will be easier starting from 0.5.0 upgrade.

**Pull request closed without merging**